azure/services/networkinterfaces/spec.go

```diff
@@ -150,6 +156,7 @@ func (s *NICSpec) Parameters(existing interface{}) (parameters interface{}, err
 		InterfacePropertiesFormat: &network.InterfacePropertiesFormat{
 			EnableAcceleratedNetworking: s.AcceleratedNetworking,
 			IPConfigurations:            &ipConfigurations,
+			DNSSettings:                 &dnsSettings,
```

Comment:
Sorry to confuse, please ignore, I think I answered my own question as I was thinking over the returned `network.Interface` object.

I had meant to ask "if we are going to break anything later on if `&dnsSettings` is a `nil` struct in the below return type? 
Should we be appending an empty slice instead to `&dnsSettings.DNSServers` ?"

But I realized that `&dnsSettings` can be nil if `len(s.DNSServers) == 0`. 

I.e. below returned object is acceptable.
```
network.Interface{
		Location: to.StringPtr(s.Location),
		InterfacePropertiesFormat: &network.InterfacePropertiesFormat{
			EnableAcceleratedNetworking: s.AcceleratedNetworking,
			IPConfigurations:            &ipConfigurations,
			DNSSettings:                 &dnsSettings, // <- this gets evaluated to <nil> if len(s.DNSServers) == 0
			EnableIPForwarding:          to.BoolPtr(s.EnableIPForwarding),
		},
	}, nil
```